attestation-agent/attestation-agent/src/lib.rs

```diff
@@ -201,12 +226,20 @@ impl AttestationAPIs for AttestationAgent {
     /// Perform the initdata binding. If current platform does not support initdata
     /// binding, return `InitdataResult::Unsupported`.
     async fn bind_init_data(&self, init_data: &[u8]) -> Result<InitDataResult> {
-        self.attester.bind_init_data(init_data).await
+        let mut init_data_result = InitDataResult::Unsupported;
+
+        for (tee, attester) in &self.attesters {
+            match attester.bind_init_data(init_data).await? {
```

Comment:
Probably initdata is similar to runtime attestation, s.t. only one (cpu/basic/platform) supports that .

Reply:
Yes. Although I am thinking that we should introduce init-data at the scope of the AA rather than attach it to TEE evidence; as in, add it as a field in the attestation message of the RCAR. I think we maybe should do something similar for the eventlog, but we can worry about that later.

This is related to the hierarchical question. Rather than have one root TEE which keeps track of the init-data and event log, I think it makes more sense to have that info at the scope of the AA, and then allow attesters and verifiers to use the info if they want to.

Reply:
Yes, here it actually just means choosing a tee to bind or check the initdata hash, as for the initdata hash or plaintext itself, it is still processed by AA. I think this is consistent with your idea. A tee "specifically marked" TEE is responsible for that.